### ISSUE TITLE
Recognize Unity's built-in Android SDK

### ISSUE THREAD

**Issue by Dentrax** [facebook/facebook-sdk-for-unity#439]
## Checklist

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-sdk-for-unity/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

- Unity Editor Version: `2019.3.3f1`
- Unity SDK Version: `I don't know what it means, never heard of 'Unity SDK', It's just Unity.`
- Installation Platform & Verison: `[Android]` version `Q - API 21`
- Facebook SDK Version: `7.19.2`

## Goals

Simply build the project without any troubleshoot

## Expected Results

It should recognize Unity's built-in SDK path if it is correct. Which is like:  `C:/Program Files/Unity/Hub/Editor/2019.3.3f1/Editor/Data/PlaybackEngines/AndroidPlayer\SDK`

Eventually, project should be successfully compiled.

## Actual Results

Image:
![Annotation 2020-05-13 215346](https://user-images.githubusercontent.com/16493751/81858513-89da1980-956c-11ea-8051-019e176b1d89.png)

## Steps to Reproduce

1. Create a new Unity `2019.3` project
2. Import the latest Facebook SDK `7.19.2`
3. Use Unity's default External Tools
4. Switch to Android
5. Build the project

## Refs

Ref: https://forum.unity.com/threads/fixed-ucb-team-android-builds-with-the-facebook-sdk-are-failing-17th-march.461561/

Ref: https://forum.unity.com/threads/you-dont-have-android-sdk-setup-facebbok-sdk.835435/

Thanks!

**Comment by rthery** [comment]
As a workaround, I created a menu entry in the Unity Editor to fix that:
https://gist.github.com/rthery/5d734b08cf36bdbaa5ccdb9b9b53b818

Based on https://github.com/facebook/facebook-sdk-for-unity/blob/9389c5cfcaf1daa2a49fac39570e51b4563f37be/Facebook.Unity.Editor/android/FacebookAndroidUtil.cs#L104

**Comment by binouze** [comment]
I have the same problem each time I update Unity.
You can uncheck the "xxx installed with unity" checkbox, correct the url (for me the unity version) of the jdk, sdk ... and then recheck the check box.

**Comment by Novack** [comment]
Getting the same issue with Unity 2020.1

**Comment by Andreyul** [comment]
It looks like they've fixed this in 7ad0cb94585c84fa53c565d53f03981cc056970d

**Comment by nick-dodonov** [comment]
> It looks like they've fixed this in [7ad0cb9](https://github.com/facebook/facebook-sdk-for-unity/commit/7ad0cb94585c84fa53c565d53f03981cc056970d)

@Andreyul 
you are wrong - fix works only when you include plugin as sources - because there's no `UNITY_2019_1_OR_NEWER` during .dll build which is provided in .unitypackage

**Comment by GilbertoBitt** [comment]
this is still a thing till today OMG! facebook just update already!

**Comment by farahjr** [comment]
> this is still a thing till today OMG! facebook just update already!

May 2021, i'm coming from the future to tell You. This is still a thing i can't work around it, since i'm not familiar to those Android Dev Ops. Tried every Unity Version i have to avoid messsing around with JDKs......and folders.

**Comment by Dentrax** [comment]
> > this is still a thing till today OMG! facebook just update already!
> 
> May 2021, i'm coming from the future to tell You. This is still a thing i can't work around it, since i'm not familiar to those Android Dev Ops. Tried every Unity Version i have to avoid messsing around with JDKs......and folders.

Is this still happening in latest version of Unity and Plugin?

cc: @KylinChang

**Comment by farahjr** [comment]
> > > this is still a thing till today OMG! facebook just update already!
> > 
> > 
> > May 2021, i'm coming from the future to tell You. This is still a thing i can't work around it, since i'm not familiar to those Android Dev Ops. Tried every Unity Version i have to avoid messsing around with JDKs......and folders.
> 
> Is this still happening in latest version of Unity and Plugin?
> 
> cc: @KylinChang

It happened with the latest version of unity 2021 and SDK 9.1 or 9.25, not completely sure since 9.2 released same day i tried to solve this.

I found a away, unchecking the boxes and pasting the same value it has, solved the problem to me.
